Gems/PhysX/Code/Source/StaticRigidBodyComponent.cpp

```diff
@@ -197,4 +226,157 @@ namespace PhysX
         return AzPhysics::SceneQueryHit();
     }
 
+    AZ::Vector3 StaticRigidBodyComponent::GetCenterOfMassWorld() const
+    {
+        return AZ::Vector3::CreateZero();
+    }
+
+    AZ::Vector3 StaticRigidBodyComponent::GetCenterOfMassLocal() const
+    {
+        return AZ::Vector3::CreateZero();
+    }
+
+    AZ::Matrix3x3 StaticRigidBodyComponent::GetInertiaWorld() const
+    {
+        return AZ::Matrix3x3::CreateIdentity();
+    }
+
+    AZ::Matrix3x3 StaticRigidBodyComponent::GetInertiaLocal() const
+    {
+        return AZ::Matrix3x3::CreateIdentity();
+    }
+
+    AZ::Matrix3x3 StaticRigidBodyComponent::GetInverseInertiaWorld() const
+    {
+        return AZ::Matrix3x3::CreateIdentity();
+    }
+
+    AZ::Matrix3x3 StaticRigidBodyComponent::GetInverseInertiaLocal() const
+    {
+        return AZ::Matrix3x3::CreateIdentity();
+    }
+
+    float StaticRigidBodyComponent::GetMass() const
+    {
+        return 0.0f;
+    }
+
+    float StaticRigidBodyComponent::GetInverseMass() const
+    {
+        return 0.0f;
+    }
+
+    void StaticRigidBodyComponent::SetMass(float)
+    {
+    }
+
+    void StaticRigidBodyComponent::SetCenterOfMassOffset(const AZ::Vector3&)
+    {
+    }
+
+    AZ::Vector3 StaticRigidBodyComponent::GetLinearVelocity() const
+    {
+        return AZ::Vector3::CreateZero();
+    }
+
+    void StaticRigidBodyComponent::SetLinearVelocity(const AZ::Vector3&)
+    {
+    }
+
+    AZ::Vector3 StaticRigidBodyComponent::GetAngularVelocity() const
+    {
+        return AZ::Vector3::CreateZero();
+    }
+
+    void StaticRigidBodyComponent::SetAngularVelocity(const AZ::Vector3&)
+    {
+    }
+
+    AZ::Vector3 StaticRigidBodyComponent::GetLinearVelocityAtWorldPoint(const AZ::Vector3&) const
+    {
+        return AZ::Vector3::CreateZero();
+    }
+
+    void StaticRigidBodyComponent::ApplyLinearImpulse(const AZ::Vector3&)
+    {
+    }
+
+    void StaticRigidBodyComponent::ApplyLinearImpulseAtWorldPoint(const AZ::Vector3&, const AZ::Vector3&)
+    {
+    }
+
+    void StaticRigidBodyComponent::ApplyAngularImpulse(const AZ::Vector3&)
+    {
+    }
```

Comment:
question: Should we emit a warning in any of these calls in case someone tries to call them?

Reply:
I thought about it and probably not worth it, they are private functions in the class, so no users can call them directly, and via Bus it's actually not bad to do a no-op implementation.